pool/web/handler/admin_experiments_assignments.ml

```diff
@@ -22,65 +22,142 @@ let index req =
+      | DeletedAssignments -> path "assignments/deleted"
+      | Session ->
+        Format.asprintf "sessions/%s" (session_id |> Id.value) |> path
+    in
+    CCList.assoc ~eq:CCString.equal Message.Field.(show Redirect) urlencoded
+    |> CCList.head_opt
+    >>= Page.Admin.Assignment.read_assignment_redirect
+    >|= to_path
+    |> value ~default:experiment_path
+  in
+  Lwt.return (session_id, assignment_id, redirect)
+;;
+
 let cancel req =
   let open Utils.Lwt_result.Infix in
-  let experiment_id, id =
-    let open Pool_common.Message.Field in
-    ( HttpUtils.find_id Experiment.Id.of_string Experiment req
-    , HttpUtils.find_id Assignment.Id.of_string Assignment req )
-  in
-  let redirect_path =
-    Format.asprintf
-      "/admin/experiments/%s/assignments"
-      (Experiment.Id.value experiment_id)
+  let%lwt _, assignment_id, redirect_path = ids_and_redirect_from_req req in
+  let result { Pool_context.database_label; _ } =
+    Utils.Lwt_result.map_error (fun err -> err, redirect_path)
+    @@ let* assignment = Assignment.find database_label assignment_id in
+       let* session =
+         Session.find_by_assignment database_label assignment.Assignment.id
+       in
+       let tags = Pool_context.Logger.Tags.req req in
+       let events =
+         Cqrs_command.Assignment_command.Cancel.handle
+           ~tags
+           (assignment, session)
+         |> Lwt.return
+       in
+       let handle events =
+         let%lwt () =
+           Lwt_list.iter_s (Pool_event.handle_event ~tags database_label) events
+         in
+         Http_utils.redirect_to_with_actions
+           redirect_path
+           [ Message.set
+               ~success:[ Pool_common.Message.(Canceled Field.Assignment) ]
+           ]
+       in
+       events |>> handle
   in
+  result |> HttpUtils.extract_happy_path req
+;;
+
+let mark_as_deleted req =
+  let open Utils.Lwt_result.Infix in
+  let%lwt _, assignment_id, redirect_path = ids_and_redirect_from_req req in
   let result { Pool_context.database_label; _ } =
     Utils.Lwt_result.map_error (fun err -> err, redirect_path)
-    @@
-    let tags = Pool_context.Logger.Tags.req req in
-    let* assignment = Assignment.find database_label id in
-    let* session =
-      Session.find_by_assignment database_label assignment.Assignment.id
-    in
-    let events =
-      let open Cqrs_command.Assignment_command.Cancel in
-      handle ~tags (assignment, session) |> Lwt.return
-    in
-    let handle events =
-      let%lwt () =
-        Lwt_list.iter_s (Pool_event.handle_event ~tags database_label) events
-      in
-      Http_utils.redirect_to_with_actions
-        redirect_path
-        [ Message.set
-            ~success:[ Pool_common.Message.(Canceled Field.Assignment) ]
-        ]
-    in
-    events |>> handle
+    @@ let* assignments =
+         Assignment.find_with_follow_ups database_label assignment_id
+       in
+       let tags = Pool_context.Logger.Tags.req req in
```

Comment:
Same here, with the "logging" issue I moved the "tags" always to the top or right after the `@@` operator. Can we stick to one solution and be consistent?